PWGHF/Core/HfMlResponse.h

```diff
@@ -0,0 +1,212 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file HFMLResponse.h
+/// \brief Class to compute the ML response for HF-analysis selections
+/// \author Fabio Catalano <fabio.catalano@cern.ch>, CERN
+/// \author Alexandre Bigot <alexandre.bigot@cern.ch>, IPHC Strasbourg
+
+#ifndef PWGHF_CORE_HFMLRESPONSE_H_
+#define PWGHF_CORE_HFMLRESPONSE_H_
+
+#include <onnxruntime/core/session/experimental_onnxruntime_cxx_api.h>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "CCDB/CcdbApi.h"
+#include "Framework/Array2D.h"
+#include "Tools/ML/model.h"
+
+#include "PWGHF/Core/SelectorCuts.h"
+
+namespace o2::analysis
+{
+
+enum PIDStatus : uint8_t {
+  TpcAndTof = 0,
+  TpcOnly,
+  TofOnly
+};
+
+/// Tag track PID status (TPC and TOF, only TPC, only TOF)
+/// \param track is the track
+/// \return tag of PID status of the track
+template <typename T1>
+uint8_t tagPID(const T1& track)
+{
+  uint8_t tag{0};
+  if (track.hasTPC() && track.hasTOF()) {
+    SETBIT(tag, PIDStatus::TpcAndTof);
+  } else if (track.hasTPC() && !track.hasTOF()) {
+    SETBIT(tag, PIDStatus::TpcOnly);
+  } else {
+    SETBIT(tag, PIDStatus::TofOnly);
```

Comment:
Shouldn't you check `!track.hasTPC() && track.hasTOF()` as a condition of setting `TofOnly`?

Reply:
Actually, if a track has no TPC and no TOF, it does not pass the PID selections performed before the ML ones. That's why I did not write it explicitely. But I can modify it to be more explicit.